### PR TITLE
fix!: able to compile & execute multi window union query in batch mode

### PR DESCRIPTION
**Pull request by aceforeverd**
this the online part of #1807 

breaking change:
- for window agg node with `append_input`, the output schema is changed
  to `producer row + project row`, which it is `project row + producer
  row` before this commit

### CODE REVIEW

**Comment by aceforeverd** [comment]
[SQL query with one union window first and one window next cause compile fail in batch mode](https://trello.com/c/Hobzsg4T/191-sql-query-with-one-union-window-first-and-one-window-next-cause-compile-fail-in-batch-mode)

File: cases/function/window/test_window_union.yaml [hunk 1]
```diff
@@ -1900,4 +1901,298 @@ cases:
...
+      order: id
+      data: |
+        1, 2, 999, 21, 221, NULL
+        2, 3, 10000, 21, 10021, 10000
+  - id: 40
+    mode: cluster-unsupport
+    desc: |
+      mulpile window union with last join
+    # FIXME(ace): fail to resolve column g2
+    tags: ["TODO"]
```
Comment (aceforeverd) [inline cases/function/window/test_window_union.yaml:2140]: left to #2563

**Comment by aceforeverd** [comment]
batch tests may fail due to this. May separate the PR into two.

File: hybridse/src/vm/physical_op.cc [hunk 2]
```diff
@@ -811,6 +828,42 @@ Status PhysicalWindowAggrerationNode::InitJoinList(PhysicalPlanContext* plan_ctx
...
+    }
+
+    // verify producer and union source has the same schema, two situation considered:
+    // 1. producer is window agg node, for batch mode, multiple window ops are serialized, where
+    //    each producer window op outputs its producer row + project row. In this case, it expect
+    //    producer schema starts with union schema
+    // 2. otherwise, always expec producer schema equals union schema
+    if (producers_[0]->GetOpType() == kPhysicalOpProject &&
+        dynamic_cast<PhysicalProjectNode*>(producers_[0])->project_type_ == kWindowAggregation) {
+        auto s = SchemaStartWith(*producers_[0]->GetOutputSchema(), *node->GetOutputSchema());
```
Comment (aceforeverd) [inline hybridse/src/vm/physical_op.cc:849]: check producer[0] has window union

**Comment by aceforeverd** [comment]
5 window tests are turned off in offline since 1cee4d645a85ebdfb7fe0bc672ca395af7788c14. Where the two tests in `TestWindow.scala` will hang up forever, not as expected. @tobegit3hub Pls re-check and enable those tests afterwards. :)

**Review by vagetablechicken — APPROVED** [review]

**Review by tobegit3hub — APPROVED** [review]
LGTM

**Pull request merged**